src/main/java/cubicchunks/server/ServerCubeCache.java

```diff
@@ -212,15 +214,16 @@ public boolean unloadQueuedChunks() {
 		final int MaxNumToUnload = 400;
 
 		// unload cubes
-		for (int i = 0; i < MaxNumToUnload && !this.cubesToUnload.isEmpty(); i++) {
-			long cubeAddress = this.cubesToUnload.poll();
+		Iterator<CubeCoords> iter = this.cubesToUnload.iterator();
+		while (iter.hasNext()) {
+			CubeCoords coords = iter.next();
```

Comment:
You're right. But I assume limiting it would be a good idea. If you agree I'll add a simple counter and exit once the maximum number is reached. But I'll have to postpone doing so to tomorrow.